index.ts

```diff
@@ -0,0 +1,476 @@
+	pulse?: MoveFlag; // Power is multiplied by 1.5 when used by a Pokemon with the Ability Mega Launcher.
+	punch?: MoveFlag; // Power is multiplied by 1.2 when used by a Pokemon with the Ability Iron Fist.
+	recharge?: MoveFlag; // If this move is successful, the user must recharge on the following turn and cannot make a move.
+	reflectable?: MoveFlag; // Bounced back to the original user by Magic Coat or the Ability Magic Bounce.
+	snatch?: MoveFlag; // Can be stolen from the original user and instead used by another Pokemon using Snatch.
+	sound?: MoveFlag; // Has no effect on Pokemon with the Ability Soundproof.
+}
+
+type MoveCategory: 'Physical' | 'Special' | 'Status';
+
+type MoveTarget =
+	// single-target
+	'normal'|'any'|'adjacentAlly'|'adjacentFoe'|'adjacentAllyOrSelf'|
+	// single-target, automatic
+	'self'|'randomNormal'|
+	// spread
+	'allAdjacent'|'allAdjacentFoes'|
+	// side and field
+	'allySide'|'foeSide'|'all';
+
+interface SelfEffect {
+	volatileStatus?: string;
+	boosts?: Partial<BoostsTable>;
+}
+
+interface SecondaryEffect {
+	chance?: number;
+	boosts?: Partial<BoostsTable>;
+	self?: SelfEffect;
+	status?: StatusName;
+	volatileStatus?: string;
+
+}
+interface Move extends Data {
+	kind: 'Move';
+
+	accuracy: number | true;
+	basePower: number;
+	flags: MoveFlags;
+	pp: number;
+	target: MoveTarget;
+	type: TypeName;
+
+	alwaysCrit?: boolean;
+	alwaysHit?: boolean;
+	basePowerModifier?: number;
+	boosts?: Partial<BoostsTable> | false
+	bypassesProtect?: boolean;
+	category: MoveCategory;
+	contestType?: string;
+	critModifier?: number;
+	critRatio?: number;
+	damage?: number | 'level' | false | null;
+	defensiveCategory?: MoveCategory;
+	drain?: [number, number];
+	forceSTAB?: boolean;
+	forceSwitch?: boolean;
+	givesHealth?: boolean;
+	hasCustomRecoil?: boolean;
+	hasPriority?: boolean;
+	hasRecoil?: Recoil;
+	hasSecondaryEffect?: boolean;
+	heal?: number[] | null;
+	ignoreAbility?: boolean;
+	ignoreAccuracy?: boolean;
+	ignoreDefensive?: boolean;
+	ignoreEvasion?: boolean;
+	ignoreImmunity?: boolean | {[k: string]: boolean};
+	ignoreNegativeOffensive?: boolean;
+	ignoreOffensive?: boolean;
+	ignorePositiveDefensive?: boolean;
+	ignorePositiveEvasion?: boolean;
+	ignoresBurn?: boolean;
+	ignoresDefenseBoosts?: boolean;
+	infiltrates?: boolean;
+	isFutureMove?: boolean;
+	isSpread?: boolean | 'allAdjacent';
+	isZ?: boolean | ID;
+	multiaccuracy?: boolean;
+	multihit?: number | number[];
+	multihitType?: 'parentalbond';
+	noCopy?: boolean;
+	noDamageVariance?: boolean;
+	noFaint?: boolean;
+	noMetronome?: string[];
+	noPPBoosts?: boolean;
+	noSketch?: boolean;
+	nonGhostTarget?: string;
+	ohko?: true | 'Ice' | null;
+	percentHealed?: number;
+	pressureTarget?: string;
+	priority?: number;
+	pseudoWeather?: string:
+	recoil?: [number, number];
+	secondaries?: SecondaryEffect[];
+	self?: SelfEffect;
+	selfBoost?: {boosts?: Partial<BoostsTable>};
+	selfSwitch?: ID | boolean;
+	selfdestruct?: string | boolean;
```

Comment:
what does the `string` represent?

Reply:
hmm, looks like whoever wrote this type was either unaware of type literals or lazy? it appears this can be `'always' | 'ifHit'`, and I dont actually see the `boolean` used at all.